rskj-core/src/main/java/co/rsk/net/NodeMessageHandler.java

```diff
@@ -110,7 +136,8 @@ public NodeMessageHandler(RskSystemProperties config,
                 config.bannedMinerList().stream().map(RskAddress::new).collect(Collectors.toSet())
         );
         this.messageQueueMaxSize = config.getMessageQueueMaxSize();
-        this.thread = new Thread(this, "message handler");
+        this.thread = Optional.ofNullable(thread).orElse(new Thread(this, "message handler"));
+        this.messageCounter = Optional.ofNullable(messageCounter).orElse(new MessageCounter());
```

Comment:
I chose to use _Optionals_ because they make the code clearer and match our usual practices in handling null values. Since NodeMessageHandler is created just once with RSKContext, there won’t be any noticeable impact on performance. It keeps things straightforward and consistent across our codebase